### PR TITLE
feat: Automatically generate a Cart ID if non provided

### LINKED ISSUE #28: Automatically generate a Cart ID if non provided
Right now `CartProvider` expects an `id` prop to function.

If no `id` is provided, we should automatically generate one, but we should also still accept custom cart IDs if someone wants to manage that themselves.

### PR DESCRIPTION
**Pull request by halafy-ds**
Closes #28

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -205,3 +217,10 @@ export function CartProvider({
     </CartContext.Provider>
   );
 }
+
+CartProvider.propTypes = propTypes
+
+CartProvider.defaultProps = {
+  id : Math.random().toString(36).substring(7),
```
Comment (notrab) [inline src/index.js:224]: We could move this up above when destructuring `CartProvider` props.

File: src/index.js [hunk 2]
```diff
@@ -85,6 +86,17 @@ const calculateTotalItems = (items) =>
...
+    defaultItems : PropTypes.array.isRequired,
+    onSetItems : PropTypes.func,
+    onItemAdd: PropTypes.func,
+    onItemUpdate: PropTypes.func,
+    onItemRemove: PropTypes.func,
+}
+
 export function CartProvider({
   children,
   id,
```
Comment (notrab) [inline src/index.js:102]:
```suggestion
  id: Math.random().toString(36).substring(7),
```

File: src/index.js [hunk 3]
```diff
@@ -205,3 +217,10 @@ export function CartProvider({
     </CartContext.Provider>
   );
 }
+
+CartProvider.propTypes = propTypes
+
+CartProvider.defaultProps = {
+  id : Math.random().toString(36).substring(7),
+  defaultItems : [],
```
Comment (notrab) [inline src/index.js:225]: This is already set above when destructuring `CartProvider` props.

File: src/index.js (hunk 1 above)
Comment (halafy-ds) [reply src/index.js]: Of course. no need of proptypes also. Just give this default value when destructuring

File: README.md [hunk 4]
```diff
@@ -113,7 +113,7 @@ ReactDOM.render(
 
 #### Props
 
-- `id` (**Required**): The `id` for your cart
+- `id`: Optional `id` for your cart to help you identify it acrcoss local storage
```
Comment (notrab) [inline README.md:116]:
```suggestion
- `id`: (_optional_) `id` for your cart to enable automatic cart retrieval via `window.localStorage`
```

File: src/index.js [hunk 5]
```diff
@@ -87,17 +87,13 @@ const calculateUniqueItems = (items) => items.length;
 
 export function CartProvider({
   children,
-  id,
+  id = Math.random().toString(36).substring(7),
```
Comment (notrab) [inline src/index.js:90]: Switching to a longer string

```suggestion
  id = [...Array(12)]
    .map((i) => (~~(Math.random() * 36)).toString(36))
    .join(""),
```

**Pull request merged**